core/imds/src/main/java/software/amazon/awssdk/imds/internal/StringResponseHandler.java

```diff
@@ -33,9 +33,9 @@
 public class StringResponseHandler implements HttpResponseHandler<String> {
     private static final Logger log = Logger.loggerFor(StringResponseHandler.class);
 
-    private CompletableFuture<String> future;
+    private CompletableFuture<?> future;
```

Comment:
Hmm, is there any reason we change it to wildcard?

Reply:
since I'm reusing the `StringResponseHandler` in the `TokenResponseHandler`, this class needs to handle CompletableFutures of String and Token. Since it's only completed exceptionnaly here, a wildcard seemed fine.

Reply:
I'm a bit confused of how this future works here. We can probably keep it as is and follow up on this in separate PR.

Reply:
Looking at the AsyncResponseHandler class, we should be able to throw an exception here instead of exceptionally completing the future. That will make things more clear